### DISCUSSION TITLE
500 error with `'NoneType' object is not iterable`

### DISCUSSION THREAD

**Discussion by LegNeato** [FlareSolverr/FlareSolverr#1041] [Q&A]
### Have you checked our README?

- [X] I have checked the README

### Have you followed our Troubleshooting?

- [X] I have followed your Troubleshooting

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you checked the discussions?

- [X] I have read the Discussions

### Environment

```markdown
- FlareSolverr version: 716efe2e0ff0
- Last working FlareSolverr version: never tried before
- Operating system: mac os
- Are you using Docker: [yes/no] yes
- FlareSolverr User-Agent (see log traces or / endpoint):
{"msg": "FlareSolverr is ready!", "version": "3.3.13", "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"}%       

- Are you using a VPN: [yes/no] no
- Are you using a Proxy: [yes/no] no
- Are you using Captcha Solver: [yes/no] no
- If using captcha solver, which one:
- URL to test this issue:

https://www.point2homes.com/US/Real-Estate-Listings/PR.html
```

### Description

Calling this in the browser console causes FlareSolverr to 500 err  with `'NoneType' object is not iterable` (as viewed in the container logs).

```js
             let body = { 
                    "cmd": "request.get",
                    "url": `https://www.point2homes.com/US/Real-Estate-Listings/PR.html`,
                };

                fetch(
                    `http://localhost:8191/v1`,
                    {  mode:  'no-cors',
                       method: 'post',
                       body }).await;
```

Running the same thing with google.com works fine.

### Logged Error Messages

```text
'NoneType' object is not iterable
```

**Comment by LegNeato** [comment]
Why was this moved to discussions @ilike2burnthing ? This is clearly a bug. I'm happy to help debug or even put up a PR if you point me in the right direction.

**Reply by ilike2burnthing** [reply]
My bad, I meant to reply to this after I moved it. Guessing I got distracted.

It's working fine by bash and python on Windows and Docker, so I'm assuming this is an issue with the console command you're running, rather than with FlareSolverr.

That all aside though, I'm not getting a Cloudflare challenge when accessing the URL in browser or with FlareSolverr. Are you?

**Reply by LegNeato** [reply]
> My bad, I meant to reply to this after I moved it. Guessing I got distracted.

No worries!
> 
> It's working fine by bash and python on Windows and Docker, so I'm assuming this is an issue with the console command you're running, rather than with FlareSolverr.

It is not. I am running the exact docker command in the README, and the `fetch` request I pasted above completes succesfully when the url is google. So something point2homes returns causes FlareSolverr to error out / 500.

> 
> That all aside though, I'm not getting a Cloudflare challenge when accessing the URL in browser or with FlareSolverr. Are you?

The first time I load a page it works fine, subsequent loads push me to cloudflare.

**Comment by LegNeato** [comment]
I was able to repro. This happens when `Content-Type: application/json` is not set. A good enhancement would be to actually log this in the 500 response.

**Comment by ilike2burnthing** [comment]
> It is not. I am running the exact docker command in the README, and the `fetch` request I pasted above completes succesfully when the url is google. So something point2homes returns causes FlareSolverr to error out / 500.

Replacing `https://www.point2homes.com/US/Real-Estate-Listings/PR.html` with `https://www.google.com/` isn't working for me, just results in the same error.

>This happens when Content-Type: application/json is not set.

Even with `headers: {'Content-Type': 'application/json'},` to the bracketed section of the fetch command, I'm still getting the error. I had tested that before as well, got the same result. I also tried rewriting the command as per the example here - https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#supplying_request_options

**Comment by janemba** [comment]
Hello,

I got the exact same issue today on yggtorrent. For the `request.post` request if `application/x-www-form-urlencoded` is used like the documentation says I got the following error:
```
'NoneType' object is not iterable
```
Now, if I let the header with only `{'Content-Type': 'application/json'}` the request works but then what should be the format of `postData` ?

**Reply by ilike2burnthing** [reply]
https://github.com/FlareSolverr/FlareSolverr/discussions/791 help you?

**Reply by janemba** [reply]
@ilike2burnthing 

Not at all, I did the same thing. My request is the following:
```
{
  'cmd': 'request.post',
  'url': 'https://www3.yggtorrent.qa/user/login',
  'postData': 'id=someuser&pass=somepassword&ci_csrf_token=',
  'session': '1234',
  'maxTimeout': 60000, 
  'returnOnlyCookies': False
}
```

And I'm using the following content-type:
```
application/x-www-form-urlencoded
```

But as I said in another thread, that 'NoneType' error is coming from `response.json()` in `FlareSolverr.py`. That call return an empty value and the cause seems to be the `Content-Type` header.

**Reply by ilike2burnthing** [reply]
The `Content-Type` header must be `application/json` - https://github.com/FlareSolverr/FlareSolverr#usage

**Reply by janemba** [reply]
It works thanks.

FYI, your documentation is confusing: https://github.com/FlareSolverr/FlareSolverr?tab=readme-ov-file#-requestpost

**Reply by ilike2burnthing** [reply]
That's specifically for the `postData` parameter for `request.post`, not the `Content-Type` header in your command. If you can think of a better way to phrase it, please submit a PR.
